system/jhtree/ctfile.cpp

```diff
@@ -1030,7 +1056,27 @@ void CJHTreeMetadataNode::get(StringBuffer & out)
 
 void CJHTreeBloomTableNode::get(MemoryBuffer & out)
 {
-    out.append(expandedSize, keyBuf);
+    out.append(expandedSize-read, keyBuf + read);
+}
+
+__int64 CJHTreeBloomTableNode::get8()
+{
+    __int64 ret = 0;
+    assert(sizeof(ret)==8);
+    assertex(expandedSize >= read + sizeof(ret));
+    _WINCPYREV8(&ret, keyBuf + read);
+    read += sizeof(ret);
```

Comment:
Having read as a member looks slightly dubious - e.g. what if it happened to be cached and reused.

Reply:
All the code is dubious round here. But we should not be caching these nodes